src/matrixrtc/MatrixRTCSession.ts

```diff
@@ -0,0 +1,385 @@
+        const roomState = room.getLiveTimeline().getState(EventTimeline.FORWARDS);
+        if (!roomState) {
+            logger.warn("Couldn't get state for room " + room.roomId);
+            throw new Error("Could't get state for room " + room.roomId);
+        }
+        const callMemberEvents = roomState.getStateEvents(EventType.GroupCallMemberPrefix);
+
+        const callMemberships: CallMembership[] = [];
+        for (const memberEvent of callMemberEvents) {
+            const eventMemberships: CallMembershipData[] = memberEvent.getContent()["memberships"];
+            if (eventMemberships === undefined) {
+                logger.warn(`Ignoring malformed member event from ${memberEvent.getSender()}: no memberships section`);
+                continue;
+            }
+            if (!Array.isArray(eventMemberships)) {
+                logger.warn(`Malformed member event from ${memberEvent.getSender()}: memberships is not an array`);
+                continue;
+            }
+
+            for (const membershipData of eventMemberships) {
+                try {
+                    const membership = new CallMembership(memberEvent, membershipData);
+
+                    if (membership.callId !== "" || membership.scope !== "m.room") {
+                        // for now, just ignore anything that isn't the a room scope call
+                        logger.info(`Ignoring user-scoped call`);
+                        continue;
+                    }
+
+                    if (membership.isExpired()) {
+                        logger.info(
+                            `Ignoring expired device membership ${memberEvent.getSender()}/${membership.deviceId}`,
+                        );
+                        continue;
+                    }
+                    callMemberships.push(membership);
+                } catch (e) {
+                    logger.warn("Couldn't construct call membership: ", e);
+                }
+            }
+        }
+
+        callMemberships.sort((a, b) => a.createdTs() - b.createdTs());
+        logger.debug(
+            "Call memberships, in order: ",
+            callMemberships.map((m) => [m.createdTs(), m.member.userId]),
+        );
+
+        return callMemberships;
+    }
+
+    /**
+     * Return a the MatrixRTC for the room, whether there are currently active members or not
+     */
+    public static roomSessionForRoom(client: MatrixClient, room: Room): MatrixRTCSession {
+        const callMemberships = MatrixRTCSession.callMembershipsForRoom(room);
+
+        return new MatrixRTCSession(client, room, callMemberships);
+    }
+
+    private constructor(
+        private readonly client: MatrixClient,
+        public readonly room: Room,
+        public memberships: CallMembership[],
+    ) {
+        super();
+        this.setExpiryTimer();
+    }
+
+    public isJoined(): boolean {
+        return this.relativeExpiry !== undefined;
+    }
```

Comment:
What does "joined" mean in this context? - I'm asking because a matrix RTC session is a session / on-going call with N participants/members. I would assume that the session is "joined" if there is at least one participant in it, "not joined" otherwise. Or did I misunderstand the meaning of a `MatrixRTCSession`?

Reply:
Probably needed a comment, sorry - does that help?